src/main/kotlin/app/revanced/patches/youtube/misc/fix/backtoexitgesture/patch/FixBackToExitGesturePatch.kt

```diff
@@ -0,0 +1,75 @@
+package app.revanced.patches.youtube.misc.fix.backtoexitgesture.patch
+
+import app.revanced.patcher.annotation.Description
+import app.revanced.patcher.annotation.Version
+import app.revanced.patcher.data.BytecodeContext
+import app.revanced.patcher.extensions.addInstruction
+import app.revanced.patcher.fingerprint.method.impl.MethodFingerprint.Companion.resolve
+import app.revanced.patcher.patch.BytecodePatch
+import app.revanced.patcher.patch.PatchResult
+import app.revanced.patcher.patch.PatchResultSuccess
+import app.revanced.patcher.util.proxy.mutableTypes.MutableMethod
+import app.revanced.patches.youtube.misc.fix.backtoexitgesture.annotation.FixBackToExitGestureCompatibility
+import app.revanced.patches.youtube.misc.fix.backtoexitgesture.fingerprints.OnBackPressedFingerprint
+import app.revanced.patches.youtube.misc.fix.backtoexitgesture.fingerprints.RecyclerViewScrollingFingerprint
+import app.revanced.patches.youtube.misc.fix.backtoexitgesture.fingerprints.RecyclerViewTopScrollingParentFingerprint
+import app.revanced.patches.youtube.misc.fix.backtoexitgesture.fingerprints.RecyclerViewTopScrollingFingerprint
+
+@Description("Closes the app by tapping the back button from the home feed.")
+@FixBackToExitGestureCompatibility
+@Version("0.0.1")
+class FixBackToExitGesturePatch : BytecodePatch(
+    listOf(
+        OnBackPressedFingerprint,
+        RecyclerViewScrollingFingerprint,
+        RecyclerViewTopScrollingParentFingerprint,
+    )
+) {
+    override fun execute(context: BytecodeContext): PatchResult {
+        val recyclerViewScrollingFingerprint = RecyclerViewScrollingFingerprint.result!!
+        injectCall(
+            recyclerViewScrollingFingerprint.mutableMethod,
+            recyclerViewScrollingFingerprint.scanResult.patternScanResult!!.endIndex,
```

Comment:
This is duplicated three times. Why not make the signature of InjectCall accept this: `SomeFingerprint.injectCall(...)`

Reply:
Please also use a data class for the parameters register, method name, and method parameters as requested [here](https://github.com/revanced/revanced-patches/pull/1405#discussion_r1059574290).